.github/workflows/validate_pr_labels_and_release_notes.yml

```diff
@@ -4,76 +4,82 @@ on:
   pull_request:
     types: [opened, synchronize, reopened, labeled, unlabeled, edited]
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
-
 permissions:
   contents: read
-  pull-requests: write # post PR comment
+  pull-requests: write
 
 jobs:
-  get-pr-labels:
-    if: "!startsWith(github.event.pull_request.head.ref, 'release/') && !contains(github.event.pull_request.user.login, '[bot]')"
+  pr-validation:
     runs-on: ubuntu-latest
+
     steps:
-      - name: Fetch repo
+      - name: Checkout repository
         uses: actions/checkout@v5
         with:
           fetch-depth: 0
 
-      - name: Get the list of allowed pull request labels
+      - name: Get allowed PR labels
+        id: get-allowed_labels
+        env:
+          PROJECT_ROOT: ${{ github.workspace }}
         run: |
-          RED='\033[0;31m'
-          NC='\033[0m'
           FILE_NAME=.pull-requests-allowed-labels-list
           FILE_PATH=$PROJECT_ROOT/$FILE_NAME
           if [[ ! -f "$FILE_PATH" ]]; then
-             echo -e "${RED}$FILE_NAME file doesn't exits in the root of the respository${NC}"
+             echo "Error: $FILE_NAME file doesn't exist in the root of the repository"
              exit 1
-          fi 
-          echo "LABELS=$(cat $FILE_PATH),chore,improvement" >> $GITHUB_OUTPUT
-        id: get-allowed_labels
-        env:
-          PROJECT_ROOT: ${{ github.workspace }}
-
-      - name: Validate labels
-        uses: jesusvasquez333/verify-pr-label-action@v1.4.0
-        with:
-          github-token: '${{ secrets.GITHUB_TOKEN }}'
-          valid-labels: ${{ steps.get-allowed_labels.outputs.LABELS }}
+          fi
+          # Add 'chore', 'improvement', 'release' as allowed labels
+          echo "LABELS=$(cat $FILE_PATH),chore,improvement,release" >> $GITHUB_OUTPUT
 
       - name: Get PR labels
-        uses: joerick/pr-labels-action@v1.0.9
         id: pr-labels
+        uses: joerick/pr-labels-action@v1.0.9
 
-      - run: |
-          ALLOWED_LABELS="$ALLOWED_LABELS,chore,improvement"
-          IFS=',' read -r -a ALLOWED_LABELS_ARRAY <<< "$ALLOWED_LABELS"
+      - name: Add release label
+        if: startsWith(github.event.pull_request.head.ref, 'release/')
+        uses: actions-ecosystem/action-add-labels@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          labels: "release"
 
-          SHOULD_FAIL_JOB="false"
+      - name: Auto-approve release PR
+        if: startsWith(github.event.pull_request.head.ref, 'release/')
+        uses: hmarr/auto-approve-action@v2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Validate release notes per label
+        if: "!startsWith(github.event.pull_request.head.ref, 'release/')"
```

Comment:
this might be misleading in the future, as it reads as semantics conflict - we do not check release notes for release PRs. I guess the only way to make it less counterintuitive is to add comments or expand the `name` section